modules/nrfxlib/nrf_802154/sl/platform/nrf_802154_platform_sl_lptimer_zephyr.c

```diff
@@ -201,3 +246,108 @@ uint64_t nrf_802154_platform_sl_lptimer_sync_lpticks_get(void)
 {
 	return m_sync_timer.target_time;
 }
+
+nrf_802154_sl_lptimer_platform_result_t nrf_802154_platform_sl_lptimer_hw_task_prepare(
+	uint64_t fire_lpticks,
+	uint32_t ppi_channel)
+{
+	bool done_on_time;
+	nrf_802154_sl_mcu_critical_state_t mcu_cs_state;
+
+	if (!hw_task_state_set(HW_TASK_STATE_IDLE, HW_TASK_STATE_SETTING_UP)) {
+		/* The only one available set of peripherals is already used. */
+		return NRF_802154_SL_LPTIMER_PLATFORM_NO_RESOURCES;
+	}
+
+	/* For nRF52, the channel allocation could be in nrf_802154_platform_sl_timer_hw_task_init.
+	 * For nRF53, the channel allocation must be here, as it would run out of resources during
+	 * initialization - some other module temporarily needs an rtc channel to initialize.
+	 */
+	m_hw_task.chan = z_nrf_rtc_timer_chan_alloc();
+	if (m_hw_task.chan < 0) {
+		hw_task_state_set(HW_TASK_STATE_SETTING_UP, HW_TASK_STATE_IDLE);
+		return NRF_802154_SL_LPTIMER_PLATFORM_NO_RESOURCES;
+	}
```

Comment:
Would it make sense to allocate the RTC channel once and then keep it forever? From what I understand the other module uses the RTC temporarily at boot and then never again.

Reply:
Good idea, it is simpler and more robust. I applied the suggestion